server/src/main/java/keywhiz/service/providers/ClientAuthFactory.java

```diff
@@ -134,36 +177,56 @@ public Client provide(ContainerRequest containerRequest,
     }
   }
 
-  private Client authorizeClientFromXfccHeader(XfccSourceConfig xfccConfig,
-      List<String> xfccHeaderValues, Principal requestPrincipal,
+  private Client authenticateClientFromForwardedData(XfccSourceConfig xfccConfig,
```

Comment:
In line 155, since the comment says that `authenticateClientFromForwardedData` uses either the xfcc header or caller-ID header to identify the client, it might be good to repeat that in Java doc comments for `authenticateClientFromForwardedData`.

And by "caller-ID header", do we always mean "caller Spiffe ID header" (which is what we use in the method)?